driver-core/src/main/com/mongodb/client/model/Aggregates.java

```diff
@@ -650,6 +656,46 @@ public static Bson sample(final int size) {
         return new SetWindowFieldsStage<>(partitionBy, sortBy, output);
     }
 
+    /**
+     * Creates a {@code $search} pipeline stage supported by MongoDB Atlas.
+     * You may use the {@code $meta: "searchScore"} expression, e.g., via {@link Projections#metaSearchScore(String)},
+     * to extract the relevance score assigned to each found document.
+     * <p>
+     * {@link Filters#text(String, TextSearchOptions)} is a legacy text search alternative.</p>
+     *
+     * @param operator A search operator.
+     * @param options Optional {@code $search} pipeline stage fields.
+     * Specifying {@code null} is equivalent to specifying {@link SearchOptions#defaultSearchOptions()}.
+     * @return The {@code $search} pipeline stage.
+     *
+     * @mongodb.atlas.manual atlas-search/query-syntax/#-search $search
+     * @mongodb.atlas.manual atlas-search/operators-and-collectors/#operators Search operators
+     * @mongodb.atlas.manual atlas-search/scoring/ Scoring
+     * @since 4.6
+     */
+    public static Bson search(final SearchOperator operator, @Nullable final SearchOptions options) {
```

Comment:
Method overloads are nice, but only if the documentation does not have to be duplicated. In our case, the decision was to always duplicate the docs, which I see as being harmful to both us (we pollute the sources with duplicates that are problematic to maintain in a consistent state) and users (the user has no way of quickly seeing what is the same and what is different in the documentation of two program elements when that documentation is either fully duplicated, or is different only for a single parameter, as would have been in this case).

This is the reason I do not want to create overloads here.

Reply:
I would go for consistency with the CRUD API and add the overload. 

This API is about being easier for users, by being less verbose than just using documents. Being convenient for users is important and we'll have to handle the extra documentation.

Reply:
> Being convenient for users is important 

I am arguing that by copying documentation in multiple places we are not providing convenient documentation for users, because users have not quick and easy way to see the difference / equivalence of multiple methods that copy each other's documentation and potentially (but not necessary) change something there.

> we'll have to handle the extra documentation

Not only maintaining duplicated documentation is resource-consuming, but we also cannot preserve the integrity: if there is no mechanism that enforces it (and there is none), it will eventually be violated. When users look at two seemingly similar methods such that for one of them we updated the docs, but for the other we forgot, it creates a new problem for the users: which one of the methods has the correct documentation?

Since both you and Jeff requested the change, I will go ahead and introduce 2 more copies to the existing 2 copies.